crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -6431,11 +6431,11 @@ mod tests {
             world.despawn(entity);
             (MaybeLocation::caller(), world.change_tick())
         }
-        let (despawner, despawn_tick) = caller_despawn(&mut world, entity);
+        let (_despawner, despawn_tick) = caller_despawn(&mut world, entity);
 
         assert_eq!((spawner, spawn_tick), *TRACKED.get().unwrap());
         assert_eq!(
-            despawner,
+            spawner,
```

Comment:
Is this intended?

Reply:
I know we have to slightly change despawn behavior to support many-to-many relationships, but I'm not exactly sure why this test needed to be changed. Guidance welcome!